chorus_book/src/guide-choreography.md

```diff
@@ -20,6 +21,27 @@ impl Choreography for HelloWorldChoreography {
 
 `Choreography` must implement the `run` method which defines the behavior of the system. The `run` method takes a reference to an object that implements the `ChoreoOp` trait. The `ChoreoOp` trait provides choreographic operators such as `locally` and `comm`.
 
+Also, each `Choreography` has an associated type `L`, which is the the list of `ChoreographyLocation`s it can operate on. You'll get a compile error if you try to work with a `ChoreographyLocation` that is not a member of `L`. To build a set of locations, you can use the macro `hlist!`.
```

Comment:
A couple of things:

1. We want to say it is a "set" not "list". We're using lists as an underlying data structure but it behaves like a set. I will rename `HList`, `hlist!`, etc. to something like `LocationSet` and `makeLocationSet` later.
2. I think we should not put this invalid example here because we have not introduced the `locally` operator yet. I would move the middle sentence and failing example to the bottom of this page.